File: frontend/src/objects/system/app/SystemMetrics.ts
// Business note: frontend mirror of backend system object/support code used by protocol and app state boundaries.
import type { EntityCount, RatingValue } from '@/objects/core/SharedObjects'

export type SystemMetrics = {
  totalOrders: EntityCount
  activeOrders: EntityCount
  resolvedTickets: EntityCount
  averageRating: RatingValue
}

File: backend/src/main/scala/system/app/objects/SystemMetrics.scala
package system.app.objects

// Business note: system-owned application object; mirror it in the frontend only when it is part of protocol or shared app state.
import system.objects.given
import system.objects.*

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.*

final case class SystemMetrics(
    totalOrders: EntityCount,
    activeOrders: EntityCount,
    resolvedTickets: EntityCount,
    averageRating: AverageRating,
)
object SystemMetrics:
  given Encoder[SystemMetrics] = deriveEncoder
  given Decoder[SystemMetrics] = deriveDecoder
